extensions/vault/model/src/main/java/io/quarkus/vault/runtime/client/dto/pki/VaultPKIGenerateCertificateBody.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.vault.runtime.client.dto.pki;
+
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.quarkus.vault.runtime.client.dto.VaultModel;
+
+public class VaultPKIGenerateCertificateBody implements VaultModel {
+
+    @JsonProperty("common_name")
+    public String subjectCommonName;
+
+    @JsonProperty("alt_names")
+    public String subjectAlternativeNames;
+
+    @JsonProperty("ip_sans")
+    public String ipSubjectAlternativeNames;
+
+    @JsonProperty("uri_sans")
+    public String uriSubjectAlternativeNames;
+
+    @JsonProperty("other_sans")
+    public List<String> otherSubjectAlternativeNames;
+
+    @JsonProperty("ttl")
+    public String timeToLive;
+
+    public String format = "pem";
+
+    @JsonProperty("private_key_format")
+    public String privateKeyFormat = "pkcs8";
```

Comment:
See above explicit format explanation.  Also, recognize that "pkcs8" is another name for "pem".

Also, this would be very cumbersome and create an ambiguous interface. To allow something to be "PEM" or "DER" we would have 2 choices...

#### 1. API returns `java.lang.Object`
The API could return an `Object` for the relevant properties or return values.  If the user doesn't specify an explicit format then  user would have to check for `byte[]` or `String` or rely on Vault's inconsistent format defaults.

#### 2. API returns `java.lang.String`
The API could return a `String` for the relevant properties or return values.  This would mean the user would have no way of easily determining the output format via Java (values would be Base64 or PEM) and would always have to specify an explicit format or rely on Vault's inconsistent format defaults.

The above options are cumbersome and since we "can", I would prefer we provide a consistent interface that is easier to use.

Note that originally it was implemented as option 1 (we returned `Object`). The current API was a refinement after we started using the API in real code.

Reply:
> The above options are cumbersome and since we "can", I would prefer we provide a consistent interface that is easier to use.

ok then. so we need to make sure that defaults are explictly specified of gthe public api, in particular when they differ from the vault's defaults.

Reply:
Are you suggesting via  method naming or some other method of making it "explicit"? All of the methods that return format selectable items are currently documented as returning "PEM".